[Go]:#2037 Completing structs concept exercise (#2121)
exercism/v3: languages/go/exercises/concept/structs/.docs/hints.md

## General

> A `structs` is a sequence of named elements, called fields, each one having a name and a type.

Reviewer: ```suggestion
A `struct` is a sequence of named elements, called fields, each one having a name and a type.
```

### 1. Creating a new remote controlled car

> - Define a new [struct][struct] with 3 fields

Reviewer: ```suggestion
- Instantiate a new [struct][struct] and fill the fields except `distance`.
```

> - Create a function that returns the type of the new created struct and updates the fields accordingly

Reviewer: ```suggestion
- Create a function that returns the type of the newly created struct and update the fields accordingly
```

### 2. Creating a race track

- Define a new [struct][struct] with 1 field
- Create a function that returns the type of the new created struct and updates the fields accordingly

### 3. Drive the car

> - Add a new field to keep track of the distance drive

Reviewer: ```suggestion
- Add a new field to keep track of the distance driven
```

- You need to check if there is enough battery to drive the car

### 4. Check if a remote controlled car can finish a race

- You need to calculate the maximum distance a car can drive

> - We assume that the car has initially the battery 100% charged

Reviewer: ```suggestion
- We assume that the cars' battery is charged to 100% initially
```
